go-controller/pkg/ovn/gateway.go

```diff
@@ -143,67 +143,66 @@ func WithLoadBalancerGroups(routerLBGroup, clusterLBGroup, switchLBGroup string)
 	}
 }
 
-// cleanupStalePodSNATs removes SNATs against nodeIP for the given node if the SNAT.logicalIP isn't an active podIP on this node
-// We don't have to worry about missing SNATs that should be added because addLogicalPort takes care of this for all pods
-// when RequestRetryObjs is called for each node add.
-// This is executed only when disableSNATMultipleGateways = true
-// SNATs configured for the join subnet are ignored.
-//
-// NOTE: On startup libovsdb adds back all the pods and this should normally update all existing SNATs
-// accordingly. Due to a stale egressIP cache bug https://issues.redhat.com/browse/OCPBUGS-1520 we ended up adding
-// wrong pod->nodeSNATs which won't get cleared up unless explicitly deleted.
+// cleanupStalePodSNATs removes pod SNATs against nodeIP for the given node if
+// the SNAT.logicalIP isn't an active podIP, the pod network is being advertised
+// on this node or disableSNATMultipleGWs=false. We don't have to worry about
+// missing SNATs that should be added because addLogicalPort takes care of this
+// for all pods when RequestRetryObjs is called for each node add.
+// Other non-pod SNATs like join subnet SNATs are ignored.
+// NOTE: On startup libovsdb adds back all the pods and this should normally
+// update all existing SNATs accordingly. Due to a stale egressIP cache bug
+// https://issues.redhat.com/browse/OCPBUGS-1520 we ended up adding wrong
+// pod->nodeSNATs which won't get cleared up unless explicitly deleted.
 // NOTE2: egressIP SNATs are synced in EIP controller.
-// TODO (tssurya): Add support cleaning up even if disableSNATMultipleGWs=false, we'd need to remove the perPod
-// SNATs in case someone switches between these modes. See https://github.com/ovn-org/ovn-kubernetes/issues/3232
 func (gw *GatewayManager) cleanupStalePodSNATs(nodeName string, nodeIPs []*net.IPNet, gwLRPIPs []net.IP) error {
-	if !config.Gateway.DisableSNATMultipleGWs {
-		return nil
```

Comment:
so you changed this behavior. I wonder why we returned here initially, I guess it is because we assume no one would ever switch their cluster from disable snat multiple gws from on to off?

Reply:
Well, there was that TODO there
```
// TODO (tssurya): Add support cleaning up even if disableSNATMultipleGWs=false, we'd need to remove the perPod
// SNATs in case someone switches between these modes. See https://github.com/ovn-org/ovn-kubernetes/issues/3232
```
So I just went for it, it was more work not doing it.